### ISSUE TITLE
Do we need the VecType template parameter in NaiveBayesClassifier?

### ISSUE THREAD

**Issue by micyril** [mlpack/mlpack#1049]
`NaiveBayesClassifier` has the following `Train` interfaces:
```
  void Train(const MatType& data,
             const arma::Row<size_t>& labels,
             const bool incremental = true);

  template<typename VecType>
  void Train(const VecType& point, const size_t label);
```
Do we really need for the second signature to have the `VecType` template parameter? Can't we just use the alias `using VecType = arma::Col<typename MatType::value_type>`?

**Comment by rcurtin** [comment]
The only nice thing is that allowing `VecType` as a template parameter allows one to pass either sparse or dense points to this incremental `Train()` overload, but I agree, I think it would be fine to make the change you suggested.  If you want to open a PR for it we can merge it.

**Comment by rcurtin** [comment]
I think this is fixed since #1068 is merged.  If not let me know and we can reopen this.
